examples/fabric-admin/device_manager/PairingManager.cpp

```diff
@@ -0,0 +1,147 @@
+    mWindowOpener = Platform::MakeUnique<Controller::CommissioningWindowOpener>(self.mCommissioner);
+
+    if (!params->verifier.empty())
+    {
+        if (params->salt.empty())
+        {
+            ChipLogError(AppServer, "Salt is required when verifier is set");
+            return;
+        }
+
+        CHIP_ERROR err = mWindowOpener->OpenCommissioningWindow(Controller::CommissioningWindowVerifierParams()
+                                                                    .SetNodeId(params->nodeId)
+                                                                    .SetEndpointId(params->endpointId)
+                                                                    .SetTimeout(params->commissioningWindowTimeout)
+                                                                    .SetIteration(params->iteration)
+                                                                    .SetDiscriminator(params->discriminator)
+                                                                    .SetVerifier(params->verifier)
+                                                                    .SetSalt(params->salt)
+                                                                    .SetCallback(&self.mOnOpenCommissioningWindowVerifierCallback));
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(AppServer, "Failed to open commissioning window with verifier: %s", ErrorStr(err));
+        }
+    }
+    else
+    {
+        SetupPayload ignored;
+        CHIP_ERROR err = mWindowOpener->OpenCommissioningWindow(Controller::CommissioningWindowPasscodeParams()
+                                                                    .SetNodeId(params->nodeId)
+                                                                    .SetEndpointId(params->endpointId)
+                                                                    .SetTimeout(params->commissioningWindowTimeout)
+                                                                    .SetIteration(params->iteration)
+                                                                    .SetDiscriminator(params->discriminator)
+                                                                    .SetSetupPIN(NullOptional)
+                                                                    .SetSalt(NullOptional)
+                                                                    .SetCallback(&self.mOnOpenCommissioningWindowCallback),
+                                                                ignored);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(AppServer, "Failed to open commissioning window with passcode: %s", ErrorStr(err));
+        }
+    }
+
+    // Clean up params
+    delete params;
+}
+
+void PairingManager::OnOpenCommissioningWindowResponse(void * context, NodeId remoteId, CHIP_ERROR err, chip::SetupPayload payload)
+{
+    PairingManager * self = static_cast<PairingManager *>(context);
+    if (self->mCommissioningWindowDelegate)
```

Comment:
In PairingManager::OnOpenCommissioningWindowVerifierResponse you are checking for nullptr, but not here. Should you be checking before using it?